stdlib/runtime/malloc.gr

```diff
@@ -135,10 +133,7 @@ let growHeap = (nbytes: WasmI32) => {
  * @param ap: WasmI32 - The pointer to free
  */
 export let free = (ap: WasmI32) => {
-  // Let memory manager know we're freeing this pointer
-  extFree(ap)
-
-  let mut blockPtr = ap - 16n // 8 bytes for malloc header + 8 bytes for GC header
+  let mut blockPtr = ap - 8n // 8 bytes for malloc header
```

Comment:
Why is the 8 bytes for GC header no longer needed?

Reply:
Ah yeah, I should have said something about this. Back in the day, this is how it worked, but it changed when I ported `malloc` to Grain. This PR restores the original behavior where this malloc doesn't (and shouldn't) have any knowledge of the GC, and the GC version of malloc (that calls this malloc) adds in the space for the GC header.